### PR TITLE
chore: Add support for Python 3.12

### PR DESCRIPTION
**Pull request by mboutet**
Really not sure that it's the only thing I need to do to support Python 3.12 😛 

Fixes https://github.com/biojppm/rapidyaml/issues/391

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -245,7 +247,7 @@ jobs:
       # use fetch-depth to ensure all tags are fetched
       - {name: checkout, uses: actions/checkout@v3, with: {submodules: recursive, fetch-depth: 0}}
       - name: create wheel
-        uses: joerick/cibuildwheel@v2.9.0
+        uses: pypa/cibuildwheel@v2.16.2
```
Comment (mboutet) [inline .github/workflows/release.yml:250]: https://github.com/joerick/cibuildwheel was relocated to https://github.com/pypa/cibuildwheel and I also bumped to the latest version which added support for Python 3.12.

**Comment by mboutet** [comment]
@biojppm, sorry for the ping, but any chance you can approve the workflows so they can run?

**Comment by mboutet** [comment]
I'm not sure if the failures in the CI are due to my changes. The failure for the `samples` workflow appears to be caused by the workflow trying to checkout commit https://github.com/biojppm/rapidyaml/pull/392/commits/c9f3f53817e08a6a7f2264dd14de6a6a0c9db0df which is from my fork rather than your repository.

The following:
 https://github.com/biojppm/rapidyaml/blob/2cf09b073db1921938f7db87cb74560512829f55/samples/fetch_content/CMakeLists.txt#L17

should be updated to support cloning from the forked repository of a given PR.

**Comment by biojppm** [comment]
Thanks for your work!

**Pull request merged**